neuralteleportation/training/teleport/pseudo.py

```diff
@@ -33,6 +36,52 @@ def simulate_teleport_distribution(model: NeuralTeleportationModel, config: "Pse
     return model.to(config.device)
 
 
+def simulate_teleportation_distribution(model: NeuralTeleportationModel,
+                                        config: "DistributionTeleportationTrainingConfig",
+                                        **kwargs) -> NeuralTeleportationModel:
+    def _get_distribution(array):
+        aux = []
+        for i in range(len(array)):
+            aux.append(array[i].item())
+        return np.array(aux)
+
+    print(f"Shifting weights to a similar distribution to a {config.cob_sampling} teleportation w/ {config.cob_range}"
+          f"COB range.")
+
+    model.cpu()
+
+    teleported_model = deepcopy(model).random_teleport(cob_range=config.cob_range,
+                                                       sampling_type=config.cob_sampling)
+
+    teleported_layers = _get_distribution(teleported_model.get_weights(concat=True))
+
+    _, aux = plt.subplots()
+    distribution, bins, _ = aux.hist(teleported_layers, 10000, density=True, cumulative=True)
+    plt.close()
```

Comment:
I don't think it's a good idea to integrate the plotting of the weights in a function whose purpose is to assign values to the weights of the network. I'd move this whole plotting of the weights' histograms to an experiment script (I assume it could be integrated in `weights_histogram.py`), and really keep this function focused on setting the weights of the network, w/o additional side-effects.

This means removing the lines targeted by comments (as well as the `_get_distribution` function a bit higher up) from this file.

Reply:
Here I'm not plotting anything. I only used pyplot to compute a cumulative density histogram of the weights in order to simulate the distribution.
Is there a better way to do this?

Reply:
Ok. I found the whole thing a bit confusing because I didn't see that the figure were saved or shown interactively anywhere, now I know why! :wink: 

Even if it might be more code in the end, I'd really suggest using a library focused on compute (like numpy), to avoid confusion like what I experienced. Numpy has a [histogram function](https://numpy.org/doc/stable/reference/generated/numpy.histogram.html), although it doesn't seem to provide the cumulative version you wanted. But a quick search lead me to [this SO thread](https://stackoverflow.com/questions/10640759/how-to-get-the-cumulative-distribution-function-with-numpy), which suggests simply using [numpy's cumsum](https://numpy.org/doc/stable/reference/generated/numpy.cumsum.html?highlight=cumsum#numpy.cumsum) after the histogram function.

I haven't looked in details as to what values you should set to all the parameters to get a similar behavior, but I'm sure you'll be able to figure this out better than I do, since you understand how you're using those bins after better than me.